core/src/main/java/org/infinispan/stream/impl/local/PersistenceEntryStreamSupplier.java

```diff
@@ -100,10 +100,11 @@ public PersistenceEntryStreamSupplier(Cache<K, V> cache, InternalEntryFactory ic
          });
          Flowable<CacheEntry<K, V>> flowable = Flowable.fromPublisher(publisher)
                .map(me -> (CacheEntry<K, V>) PersistenceUtil.convert(me, iceFactory));
-         Iterable<CacheEntry<K, V>> iterable = () -> new LazyConcatIterator<>(localIterator,
+         CloseableIterator<CacheEntry<K, V>> iterator = new LazyConcatIterator<>(localIterator,
```

Comment:
I have to look closer, but this could explain a different issue we saw before. Good catch @pferraro

Reply:
Yes - without this, the semaphore permit acquisitions in PersistenceManagerImpl.publshEntries/publishKeys() are never released.
However, this isn't sufficient to fix the problem as nothing was yet triggering the onComplete/onError handlers.